src/Access/ReplicatedAccessStorage.cpp

```diff
@@ -0,0 +1,657 @@
+        {
+            /// The UUID already exists, simply fail.
+            /// To fail with a nice error message, we need info about what already exists.
+            /// This itself could fail if the conflicting uuid disappears in the meantime.
+            /// If that happens, then we'll just retry from the start.
+            String existing_entity_definition;
+            const bool exists = zookeeper->tryGet(entity_path, existing_entity_definition);
+            if (!exists)
+                return false; // Retry
+
+            AccessEntityPtr existing_entity = parseAccessEntity(existing_entity_definition, "znode", entity_path);
+            EntityType existing_type = existing_entity->getType();
+            String existing_name = existing_entity->getName();
+            throwIDCollisionCannotInsert(id, type, name, existing_type, existing_name);
+        }
+        else if (replace_if_exists)
+        {
+            /// The name already exists for this type.
+            /// If asked to, we need to replace the existing entity.
+            String existing_entity_uuid;
+            Coordination::Stat name_stat;
+            /// First get the uuid of the existing entity
+            /// This itself could fail if the conflicting name disappears in the meantime.
+            /// If that happens, then we'll just retry from the start.
+            const bool uuid_exists = zookeeper->tryGet(name_path, existing_entity_uuid, &name_stat);
+            if (!uuid_exists)
+                return false; // Retry
+
+            /// Then get the definition
+            const String existing_entity_path = zookeeper_path + "/uuid/" + existing_entity_uuid;
+            String existing_entity;
+            Coordination::Stat entity_stat;
+            const bool name_exists = zookeeper->tryGet(existing_entity_path, existing_entity, &entity_stat);
+            if (!name_exists)
+                return false; // Retry
+
+            Coordination::Requests replace_ops;
+            replace_ops.emplace_back(zkutil::makeRemoveRequest(existing_entity_path, entity_stat.version));
+            replace_ops.emplace_back(zkutil::makeCreateRequest(entity_path, new_entity_definition, zkutil::CreateMode::Persistent));
+            replace_ops.emplace_back(zkutil::makeSetRequest(name_path, entity_uuid, name_stat.version));
+
+            Coordination::Responses replace_responses;
+            const auto replace_res = zookeeper->tryMulti(replace_ops, replace_responses);
+            if (replace_res != Coordination::Error::ZOK)
+                return false; // Retry
+        }
+        else
+        {
+            throwNameCollisionCannotInsert(type, name);
+        }
+    }
+    else if (res != Coordination::Error::ZOK)
+    {
+        /// Other codes are unexpected, will throw
+        zkutil::KeeperMultiException::check(res, ops, responses);
+        assert(false);
+        __builtin_unreachable();
+    }
+
+    refreshEntityNoLock(zookeeper, id, notifications);
```

Comment:
It's not necessary because the existing `refreshEntityNoLock` will call `setEntityNoLock` which will detect the conflicting entity and remove it from local storage.